magento/inventory
# MSI-1542: Provide MSI support for Shipment Web API endpoint

### Description
The problem is, there is no data migration from request extension attributes to shipment extension attributes. I added service for migration request extension attributes data to shipping extension attributes data

### Fixed Issues (if relevant)
1. [magento-engcom/msi-1542](https://github.com/magento-engcom/msi/issues/1542): Provide MSI support for Shipment Web API endpoint

### Manual testing scenarios
1. Go to Magento/InventoryShipping/etc/extension_attributes.xml and add extension attribute like this 
```
    <extension_attributes for="Magento\Sales\Api\Data\ShipmentCreationArgumentsInterface">
        <attribute code="source_code" type="string" />
    </extension_attributes>
```
2. Flush cache bin/magento cache:flush
3. Run command bin/magento setup:di:compile
4. Create order via storefront
5. Run POST request /V1/order/:orderId/ship
_example: http://example.com/rest/V1/order/1/ship_
with body 
`{
  "arguments": {
    "extension_attributes": {
      "source_code": "<<<your source code>>>"
    }
  }
}`
7. Go to admin panel to order and check that the shipping was created 

### Contribution checklist
